app/(api)/_datalib/collections/findItem.ts

```diff
@@ -0,0 +1,33 @@
+import { getClient } from '@utils/mongodb/mongoClient.mjs';
+import { NotFoundError } from '@utils/response/Errors';
+
+export async function findCollectionItems(collection: never, query = null) {
+  try {
+    const client = await getClient();
+    const db = client.db;
+    const reqCollection = await db.getCollection(collection);
+    const reqDocument = await reqCollection.find(query).toArray();
+    if (!reqDocument || reqDocument.length === 0) {
+      return {
```

Comment:
I think it's easier to throw an error, and have the catch block handle the return statement. If you look at examples from pokemon or the hack davis judging app code that I posted in the CMS channel resources, you can see how this is handled.